File: 백준/Silver/2217. 로프/로프.cc
#include <cstdio>
#include <algorithm>
using namespace std;
 
int main(){
    int N, w[100000];
    scanf("%d", &N);
    for(int i=0; i<N; i++)
        scanf("%d", w+i);
 
    // 로프의 강도 정렬
    sort(w, w+N);
 
    // 결과 구하기
    int result = 0;
    for(int i=1; i<=N; i++)
        result = max(result, w[N-i]*i);
    printf("%d\n", result);
}

File: 백준/Silver/2217. 로프/로프.java

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {

        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(bf.readLine());
        Integer[] lopes = new Integer[N];

        for(int i=0; i<N; i++) {
            lopes[i] = Integer.parseInt(bf.readLine());
        }

        Arrays.sort(lopes, (l1, l2)-> l2- l1);

        int max = 0;
        for(int i=0; i<N; i++) {
            max = Math.max(max, lopes[i]*(i+1));
        }

        System.out.println(max);

    }
}